egs/wsj/s5/steps/nnet3/make_cnn_tdnn_configs.py

```diff
@@ -0,0 +1,397 @@
+try:
+    for string in split1:
+        split2 = string.split(",")
+        if len(split2) != 11:
+            sys.exit("invalid --cnn-indexes argument, must contain 11 positive integers: "
+                     + args.cnn_indexes)
+        int_list = []
+        for int_str in split2:
+            if int(int_str) < 1:
+                sys.exit("non-positive number in --cnn-indexes argument: {0}".format(int(int_str)))
+            int_list.append(int(int_str))
+        cnn_array.append(int_list)
+except ValueError as e:
+    sys.exit("invalid --cnn-indexes argument " + args.cnn_indexes + str(e))
+num_cnn_layers = len(cnn_array)
+
+
+f = open(args.config_dir + "/vars", "w")
+print('left_context=' + str(left_context), file=f)
+print('right_context=' + str(right_context), file=f)
+# the initial l/r contexts are actually not needed.
+# print('initial_left_context=' + str(splice_array[0][0]), file=f)
+# print('initial_right_context=' + str(splice_array[0][-1]), file=f)
+print('num_hidden_layers=' + str(num_hidden_layers), file=f)
+f.close()
+
+f = open(args.config_dir + "/init.config", "w")
+print('# Config file for initializing neural network prior to', file=f)
+print('# preconditioning matrix computation', file=f)
+print('input-node name=input dim=' + str(args.feat_dim), file=f)
+list=[ ('Offset(input, {0})'.format(n) if n != 0 else 'input' ) for n in splice_array[0] ]
+if args.ivector_dim > 0:
+    print('input-node name=ivector dim=' + str(args.ivector_dim), file=f)
+    list.append('ReplaceIndex(ivector, t, 0)')
+# example of next line:
+# output-node name=output input="Append(Offset(input, -3), Offset(input, -2), Offset(input, -1), ... , Offset(input, 3), ReplaceIndex(ivector, t, 0))"
+print('output-node name=output input=Append({0})'.format(", ".join(list)), file=f)
+f.close()
+
+for l in range(1, num_hidden_layers + 1):
+    f = open(args.config_dir + "/layer{0}.config".format(l), "w")
+    print('# Config file for layer {0} of the network'.format(l), file=f)
+    if l == 1:
+        if args.ivector_dim > 0:
+            print('component name=ivector-affine type=NaturalGradientAffineComponent '
+                  'input-dim={0} output-dim={0}'.format(args.ivector_dim), file=f)
+
+        if args.use_mfcc == "true":    
+            # generate the IDCT matrix and write to the file
+            idct_matrix = ComputeIDctMatrix(args.feat_dim, args.feat_dim, args.cepstral_lifter)
+            # append a zero column to the matrix, this is the bias of the fixed affine component
+            zero_col = np.zeros((args.feat_dim, 1))
+            f2 = open(args.config_dir + "/idct.mat", "w")
+            print('[ ', file=f2)
+            np.savetxt(f2, np.append(idct_matrix, zero_col, 1), fmt='%.6e')
+            print(' ]', file=f2)
+            f2.close()
+
+            print('component name=idct type=FixedAffineComponent matrix={0}/idct.mat'.
+                  format(args.config_dir), file=f)
+
+        if args.add_delta == "true":
```

Comment:
You could print a warning here saying that add-deltas was not found to be beneficial in your experiments.